DOCS-9805 "Create an API Instance" document edits (#331)
mulesoft/docs-api-manager: modules/ROOT/pages/create-instance-task.adoc

include::_attributes.adoc[]
endif::[]

> 

Reviewer: edit instructions

Type: CE
Scope: Green
Due: When you can

In API Manager 2.x, instances of an API can exist in multiple environments. Additionally, you can create multiple instances of an API in the same environment. 

> An example: 

Reviewer: Provide a more robust introduction to the following example, and use a number list instead of a bullet list, because the items are chronological.

Author: I reverted this introductory section back to the original text, as I misunderstood that the section wasn't procedural.

> * Create an instance of an API and configure it as a proxy. Apply caching policies to this proxy. 

Reviewer: Make the second sentence a second action (2.), or follow the bullet list format shown in our style reference, which inserts a black line between the list item and any further detail for it....
ACTUALLY...
...looking at the rest of this, I suggest making this all one step, based on the overall task...:

Reviewer: ```suggestion
. Create an instance of an API, configure it as a proxy, and apply caching policies to this proxy. 
```

> * Then, create an API instance to manage directly as a basic endpoint. Apply throttling and security policies to this basic endpoint. 

Reviewer: ```suggestion
. Create an API instance to manage directly as a basic endpoint, and apply throttling and security policies to this endpoint. 
```

> * After configuring an API version for management, repeat the process to create another API instance. Specifying the same version creates another API instance of the version. 

Reviewer: ```suggestion
* Repeat Step 2 to create another API instance. 
+
Specifying the same version creates another API instance of the version. 
```

> * Finally, create multiple API instances and click the version information to view API details.

Reviewer: ```suggestion
. Repeat Steps 1 through 3 to create multiple API instances, and click the version information to view API details.
```

> The following procedure describes creating an API instance from Exchange, and configuring the API instance as an endpoint with proxy.

Reviewer: Move this to follow the next subtitle and introduce the task therein.

We avoid wording that creates an organizational interdependency among topics ("the following..." with reference to a separate topic).

Author: Fixed.

== Create an API instance

> 

Reviewer: Insert previous text here, ending it with a colon rather than a period.

Reviewer: Many of the following steps begin with "From," but it is more typical in our docs to see "In." GLOBAL

Author: Both fixed.

. From *API Administration*, click *Manage API*. 
. Select *Manage API From Exchange*.
+
To configure version 1.0 of an example API named `jsonplaceholder`:
+
image::manage-exchange-end-task.png[]
+
. From *API name*, enter `jsonplaceholder`.
+

> NOTE: OpenAPI Specification (OAS) 3.0 is supported, with the exception of the callback feature. To work around this issue, handle the callback outside of the Mule runtime engine domain or use an OAS 3.0 specification that does not use callbacks.

Reviewer: Try to avoid breaking up the steps of a task by inserting a notice.
Does this info belong here; is it specific to the step directly above it?
If not, move it outside of the procedure, either before or after, as most useful to the audience.

Reviewer: ```suggestion
Although OpenAPI Specification (OAS) 3.0 is supported, the callback feature is not. To work around this issue, either handle the callback outside of the Mule runtime engine domain or use an OAS 3.0 specification that does not use callbacks.
```

Author: Fixed. I added a "Notes" section at the bottom and moved this there.

+

> . From *API version*, select version (`1.0`).

Reviewer: Why the parentheses?

Author: Fixed.

. From *Asset version*, select the Exchange asset version.
. From *Managing type*, select *Endpoint with proxy*.
. In *Implementation URI*, enter your web site URL or an arbitrary URL, such as `+http://api.domain.com+`. 
+

> If you choose _Endpoint with Proxy_ and do not make an entry (such as accepting the default value), you cannot save your configuration.

Reviewer: Doesn't this belong after the step in Line 35?

Author: Fixed.

. From *Client provider*, select the client provider to authenticate API access from registered client applications:
+
.Associate client provider with the API
image::create-instance-endpoint-midp.png[65%,65%, Create instances with multiple client provider]
+

> * The first item under API Configuration is the *Add a Tag* field.

Reviewer: ```suggestion
* The first item under *API Configuration* is the *Add a Tag* field.
```

Author: Fixed.

> * The second item shows the *Managing Type* field. Select *Endpoint With Proxy*.

Reviewer: This description list item contains a hidden step that should probably be pulled out into the main step flow of this task.

Author: It really isn't an action step, but rather a check to make sure an element is enabled. I changed the wording to reflect this.

* The third item shows the *Client provider* field. 
+
If no client providers are configured for your environment and organization, the *Client provider* field is not visible.

> . Select a proxy deployment target, for example, CloudHub.

Reviewer: ```suggestion
. Select a proxy deployment target, for example, *CloudHub*.
```

Author: Fixed.

. Accept the default path `/` for the proxy endpoint or enter another path.
. In *Advanced options*, optionally assign an instance label for the proxy (or accept the default label), and set a response timeout.
. Click *Save*.
+

> . Click *API Administration* to return to the API Administration dashboard.

Reviewer: If API Administration is a label, boldface it.

Author: Fixed.

include::_attributes.adoc[]
endif::[]

In API Manager 2.x, instances of an API can exist in multiple environments. Additionally, you can create multiple instances of an API in the same environment. 

You can create one instance of an API that serves as a proxy. You can create another to manage directly as a basic endpoint. You can apply caching policies (for example, to the proxy), and throttling and security policies to the basic endpoint.

After setting up an API version for management, you repeat the process to create another instance. Attempting to configure the same version creates another instance of the version. You can create multiple instances and expand the version information to view details about them.

== Create an API instance

The following procedure describes creating a single API instance from Exchange, and configuring the API instance as an endpoint with proxy:

. In *API Administration*, click *Manage API*. 
. Select *Manage API From Exchange*.
+

> To configure version 1.0 of an example API named `jsonplaceholder`:

Reviewer 2: Keep in mind that to Create an API Instance an Exchange asset is required.
That means that in Exchange there should be a `jsonplaceholder` asset with 1.0 version.

Maybe this is worth mentioning. (The fact that pre-condition to "Create an API Instance from Exchange", requires the asset)

API Manager supports to manage these asset types: REST API - RAML, REST API - OAS, SOAP API - WSDL, HTTP API.

Author: Fixed.

+
image::manage-exchange-end-task.png[]
+

> . In *API name*, enter `jsonplaceholder`.

Reviewer 2: This would be the name of the asset

Author: Fixed.

> . In *API version*, select version `1.0`.

Reviewer 2: Once an asset selected, it will allow the user to select the version of the asset to manage.

Author: Fixed.

. In *Asset version*, select the Exchange asset version.
. In *Managing type*, select *Endpoint with proxy*.
+
If you choose _Endpoint with Proxy_ and do not make an entry (such as accepting the default value), you cannot save your configuration.
. In *Implementation URI*, enter your web site URL or an arbitrary URL, such as `+http://api.domain.com+`. 

. In *Client provider*, select the client provider to authenticate API access from registered client applications:
+
.Associate client provider with the API
image::create-instance-endpoint-midp.png[65%,65%, Create instances with multiple client provider]
+

> * The first item under *API Configuration* is the *Add a Tag* field.

Reviewer 2: Tags are optional fields, not required.

Author: Fixed.
